src/faillog.c

```diff
@@ -199,8 +202,11 @@ static bool reset_one (uid_t uid)
 	off_t offset;
 	struct faillog fl;
 
-	offset = (off_t) uid * sizeof (fl);
-	if (offset + sizeof (fl) <= statbuf.st_size) {
+	if (__builtin_mul_overflow(uid, sizeof (fl), &offset)) {
+		return false;
+	}
+
+	if (statbuf.st_size >= (off_t) sizeof (fl) && offset <= statbuf.st_size - (off_t) sizeof (fl)) {
```

Comment:
I believe the casts to `(off_t )` are unnecessary regarding overflows, which this commit is about.  It may be necessary to silence compiler warnings, but I think that would deserve a separate patch.

Reply:
Yep.  I will rarely complain about splitting commits.  :-)